src/Compilers/CSharp/Test/Symbol/Compilation/SemanticModelGetDeclaredSymbolAPITests.cs

```diff
@@ -5267,6 +5267,434 @@ void F(int x = 2, = 3) { }
+                // (4,18): warning CS8632: The annotation for nullable reference types should only be used in code within a '#nullable' annotations context.
+                //     public string? Value;
+                Diagnostic(ErrorCode.WRN_MissingNonNullTypesContextForAnnotation, "?").WithLocation(4, 18),
+                // (9,52): warning CS8632: The annotation for nullable reference types should only be used in code within a '#nullable' annotations context.
+                //     public static Thing CreateThing(int key, string? value)
+                Diagnostic(ErrorCode.WRN_MissingNonNullTypesContextForAnnotation, "?").WithLocation(9, 52),
+                // (14,13): error CS0747: Invalid initializer member declarator
+                //             Value,
+                Diagnostic(ErrorCode.ERR_InvalidInitializerElementInitializer, "Value").WithLocation(14, 13)
+                );
+
+            var tree = comp.SyntaxTrees[0];
+            var model = comp.GetSemanticModel(tree);
+            var root = tree.GetCompilationUnitRoot();
+
+            var initializers = root.DescendantNodes()
+                .OfType<InitializerExpressionSyntax>()
+                .First()
+                .ChildNodes()
+                .ToArray();
+            var initializedSymbol = model.GetSymbolInfo(initializers[1]).Symbol;
+            Assert.Equal("System.String? Thing.Value", initializedSymbol.ToTestDisplayString());
+        }
+
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/74348")]
+        public void ObjectInitializerIncompleteMemberValueAssignment04()
+        {
+            var source = """
+                public class Thing
+                {
+                    public int Key;
+                    public event Action? Handler;
+                }
+
+                public class Using
+                {
+                    public static Thing CreateThing(int key, Action? handler)
+                    {
+                        return new()
+                        {
+                            Key = key,
+                            Handler,
+                        };
+                    }
+                }
+                """;
+            var comp = CreateCompilation(source);
+
+            comp.VerifyDiagnostics(
+                // (4,18): error CS0246: The type or namespace name 'Action' could not be found (are you missing a using directive or an assembly reference?)
+                //     public event Action? Handler;
+                Diagnostic(ErrorCode.ERR_SingleTypeNameNotFound, "Action").WithArguments("Action").WithLocation(4, 18),
+                // (4,26): error CS0066: 'Thing.Handler': event must be of a delegate type
+                //     public event Action? Handler;
+                Diagnostic(ErrorCode.ERR_EventNotDelegate, "Handler").WithArguments("Thing.Handler").WithLocation(4, 26),
+                // (4,26): warning CS0067: The event 'Thing.Handler' is never used
```

Comment:
>// (4,26): warning CS0067: The event 'Thing.Handler' is never used [](http://example.com/codeflow?start=16&length=66)

Please suppress this warning with pragma

Reply:
Done